src/index.ts

```diff
@@ -106,38 +132,43 @@ class Envoy {
 
     async connections () {
 	try {
-	    const ifaces		= this.conductor_opts.interfaces;
-
-	    this.hcc_clients.master	= new WebSocket(`ws://localhost:${ifaces.master_port}`,   RPC_CLIENT_OPTS );
-	    this.hcc_clients.service	= new WebSocket(`ws://localhost:${ifaces.service_port}`,  RPC_CLIENT_OPTS );
-	    this.hcc_clients.internal	= new WebSocket(`ws://localhost:${ifaces.internal_port}`, RPC_CLIENT_OPTS );
-	    this.hcc_clients.hosted	= new WebSocket(`ws://localhost:${ifaces.hosted_port}`,   RPC_CLIENT_OPTS );
+	    const ifaces		= this.conductor_opts.interfaces;		
+	    this.hcc_clients.master	= await HcAdminWebSocket.init(`ws://localhost:${ifaces.master_port}`);
+	    this.hcc_clients.service	= await HcAppWebSocket.init(`ws://localhost:${ifaces.service_port}`);
+	    this.hcc_clients.internal	= await HcAppWebSocket.init(`ws://localhost:${ifaces.internal_port}`);
+	    this.hcc_clients.hosted	= await HcAppWebSocket.init(`ws://localhost:${ifaces.hosted_port}`);
 	} catch ( err ) {
 	    console.error( err );
 	}
 
 	Object.keys( this.hcc_clients ).map(k => {
-	    this.hcc_clients[k].name = k;
-	    this.hcc_clients[k].port = this.conductor_opts.interfaces[`${k}_port`];
-	    log.info("Conductor client '%s' configured for port (%s)", k, this.hcc_clients[k].port );
+		this.hcc_clients[k].setSocketInfo({
+			name: k,
+			port: this.conductor_opts.interfaces[`${k}_port`]
+		 });
+	    log.info("Conductor client '%s' configured for port (%s)", k, this.hcc_clients[k].checkConnection.port );
 	});
 
 	const clients			= Object.values( this.hcc_clients );
 	this.connected			= Promise.all(
 	    clients.map(async (client:any) => {
 		await client.opened( CONDUCTOR_TIMEOUT )
 		    .catch( err => {
-			log.fatal("Conductor client '%s' failed to connect: %s", client.name, String(err) );
-			console.log( client.name, err );
-		    });
-		log.debug("Conductor client '%s' is 'CONNECTED': readyState = %s", client.name, client.socket.readyState );
+			log.fatal("Conductor client '%s' failed to connect: %s", client.checkConnection.name, String(err) );
```

Comment:
why is `checkConnection` the source of truth for the connection name? Seems like either `name` should be stored in `HcAdminWebSocket` directly, or `checkConnection` should be called something else.